src/main.cpp

```diff
@@ -1,105 +1,106 @@
-#include <Arduino.h>
-#include <NativeEthernet.h>
-#include <NativeEthernetUdp.h>
+#include "EthernetDriver.h"
 
 #include "pb_encode.h"
 #include "pb_decode.h"
 #include "urc.pb.h"
 
-IntervalTimer encoderMessageTimer;
-elapsedMillis elapsedTime;
-
-byte mac[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };
-IPAddress ip(192, 168, 8, 255);
-constexpr uint16_t localPort = 8888;
-EthernetUDP Udp;
+// Function Declarations
+void setFlag();
 
+IntervalTimer timer;
 volatile bool printFlag = false;
 
-void sendEncoderEstimates(DriveEncodersMessage driveEncodersMessage);
-void updateFlag();
-void connectUdp();
+using namespace ethernet_driver;
+
+int main() {
+
+  elapsedMillis elapsedTime;
 
-void setup()
-{
   Serial.begin(9600);
   Serial.println("Hello World!");
 
-  connectUdp();
+  // Construct UDP instance
+  EthernetDriver ethernet_driver;
+  ethernet_driver.initDriver();
+  timer.begin(setFlag, 100000);
+
+  while (true) {
+    noInterrupts();
+    if (printFlag) {
+      Serial.println(elapsedTime);
+      elapsedTime = 0;
 
-  encoderMessageTimer.begin([]() { printFlag = true; }, 100000);
+      ethernet_driver.sendMessage();
+
+      printFlag = false;
+    }
+    interrupts();
+
+    delay(5);
+  }
+
+  return 0;
+}
+
+void setFlag() {
+  printFlag = true;
 }
 
-void loop()
-{
-  // create Nanopb messages
+#if defined(ARDUINO_STYLE_FUNCS)
```

Comment:
Structuring the code like this is going to make things much harder in the future. You really should just stick to one unified design methodology

Reply:
I have a bad habit of commenting out code. I was experimenting with using the preprocessor directives instead of commenting out code. Probably need to make it policy to never push commented-out code, either make it a method or delete it.